### PR TITLE
feat(payments): document redirectBefore/AfterOrder workflows

### PR DESCRIPTION
**Pull request by abailleulfc**
Document redirectBeforeOrder workflow implementation 
https://deploy-preview-490--heuristic-almeida-1a1f35.netlify.app/docs/advanced/payments/payment-workflows-specificities#redirect-before-order

### CODE REVIEW

File: docs/advanced/payments/payment-workflows-specificities.mdx [hunk 1]
```diff
@@ -158,3 +165,138 @@ const getAdditionalActionComponent = (paymentCode, paymentAdditionalData) => {
...
+
+No AdditionalDataComponent is required for this implementation as there is no
+frontend component displayed by Front-Commerce
+
+:::
+
+1. Reference the payment workflow
+
+Override `src/web/theme/pages/Checkout/checkoutFlowOf.js` to indicate the use of
+`redirectBeforeOrder`
```
Comment (dpobel) [inline docs/advanced/payments/payment-workflows-specificities.mdx:282]: that's not really a warning.

```suggestion
No AdditionalDataComponent is required for this implementation as there is no
frontend component displayed by Front-Commerce, so you just need to override
`src/web/theme/pages/Checkout/checkoutFlowOf.js` to indicate the use of
`redirectBeforeOrder`
```

File: docs/advanced/payments/payment-workflows-specificities.mdx [hunk 2]
```diff
@@ -158,3 +165,138 @@ const getAdditionalActionComponent = (paymentCode, paymentAdditionalData) => {
...
+      METHOD_TITLE,
+      loader,
+    );
+
+    return {}; // you may export loaders here in case the payment provides custom Queries (to fetch a payment token for instance)
+  },
+};
+```
+
+2. Implement the three **required** methods in the loader
```
Comment (dpobel) [inline docs/advanced/payments/payment-workflows-specificities.mdx:218]: same remark as in the previous MR, please properly a numbered list in markdown

File: docs/advanced/payments/payment-workflows-specificities.mdx [hunk 3]
```diff
@@ -158,3 +165,138 @@ const getAdditionalActionComponent = (paymentCode, paymentAdditionalData) => {
...
+
+```js
+export default class MyPaymentLoader {
+  constructor(shopConfig, methodCode, ...) {
+    ...
+
+    this.callbackUrl = `${shopConfig.url}/checkout/payment/${methodCode}`;
+  }
+
+  mapCartPaymentToRedirect({ paymentData /* TODO ? */, cart }) {
```
Comment (dpobel) [inline docs/advanced/payments/payment-workflows-specificities.mdx:228]: TODO ?

File: docs/advanced/payments/payment-workflows-specificities.mdx [hunk 4]
```diff
@@ -158,3 +165,138 @@ const getAdditionalActionComponent = (paymentCode, paymentAdditionalData) => {
...
+    // reminder : the callbackUrl to FC is calculated in this.callbackUrl (defined in the constructor)
```
Comment (dpobel) [inline docs/advanced/payments/payment-workflows-specificities.mdx:229]: I don't understand this comment. `this.callbackUrl` is not used anywhere. Is it supposed to be used in `value` below ?

**Review by dpobel — APPROVED** [review]

**Review by Kagaminara — APPROVED** [review]

**Pull request merged**